pvlib/modelchain.py

```diff
@@ -1009,26 +1009,26 @@ def _set_celltemp(self, model):
         temp_air = _tuple_from_dfs(self.results.weather, 'temp_air')
         wind_speed = _tuple_from_dfs(self.results.weather, 'wind_speed')
         kwargs = {}
-        if model == self.system.noct_sam_celltemp:
+        if model == 'noct_sam':
```

Comment:
The docstring says that `model` is a function, it's changed to be a string here and elsewhere.

With this change, is `ModelChain.temperature_model` needed?

Reply:
> With this change, is ModelChain.temperature_model needed?

I think this question remains, but it should be explored in a separate issue.